ThirdPartyAdapters/applovin/applovin/src/main/java/com/applovin/mediation/AppLovinNativeAdListener.java

```diff
@@ -0,0 +1,109 @@
+package com.applovin.mediation;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.applovin.nativeAds.AppLovinNativeAd;
+import com.applovin.nativeAds.AppLovinNativeAdLoadListener;
+import com.applovin.nativeAds.AppLovinNativeAdPrecacheListener;
+import com.applovin.sdk.AppLovinSdk;
+import com.applovin.sdk.AppLovinSdkUtils;
+import com.google.android.gms.ads.AdRequest;
+import com.google.android.gms.ads.mediation.MediationNativeListener;
+
+import java.lang.ref.WeakReference;
+import java.util.List;
+
+class AppLovinNativeAdListener
+        implements AppLovinNativeAdLoadListener, AppLovinNativeAdPrecacheListener {
+
+    private final AppLovinNativeAdapter mAdapter;
+    private final MediationNativeListener mNativeListener;
+    private final AppLovinSdk mAppLovinSdk;
+    private final WeakReference<Context> mContextWeakReference;
+
+    AppLovinNativeAdListener(AppLovinNativeAdapter adapter,
+                             MediationNativeListener nativeListener,
+                             AppLovinSdk sdk,
+                             Context context) {
+        mAdapter = adapter;
+        mNativeListener = nativeListener;
+        mAppLovinSdk = sdk;
+        mContextWeakReference = new WeakReference<>(context);
+    }
+
+    @Override
+    public void onNativeAdsLoaded(List<AppLovinNativeAd> nativeAds) {
+        if (nativeAds.size() > 0 && isValidNativeAd(nativeAds.get(0))) {
+            mAppLovinSdk.getNativeAdService().precacheResources(nativeAds.get(0), this);
+        } else {
+            Log.w(AppLovinNativeAdapter.TAG,
```

Comment:
Lets mark this as error, since theoretically our SDK will never return empty list of native ads, if it 204s, it'll just call back the `onNativeAdsFailedToLoad()` callback.